dashboard/app/models/stage.rb

```diff
@@ -124,4 +124,30 @@ def summarize
     end
     stage_summary.freeze
   end
+
+  def lockable_state(students)
+    return unless self.lockable?
+
+    # assumption that lockable selfs have a single (assessment) level
```

Comment:
All of this seems pretty specific to `LevelGroup`.  Is the plan to expand this in the future to apply to stages with other level types in them?

Otherwise a lot of this logic might make more sense on the `LevelGroup` model (probably too late for this feedback).